packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -91,6 +91,13 @@ export interface EngineReportingOptions<TContext> {
    * Minimum back-off for retries. Defaults to 100ms.
    */
   minimumRetryDelayMs?: number;
+  // Whether to include the entire document in the trace if the operation
+  // was a GraphQL parse or validation error (i.e. failed the GraphQL parse or
+  // validation phases). This will be included as a separate field on the trace
+  // and the operation name and signature will always be reported with a static
+  // identifier. Whether the operation was a parse failure or a validation
+  // failure will be embedded within the stats report key itself
+  sendOperationDocumentsOnValidationFailure?: boolean;
```

Comment:
Why not separate flags for validation and parsing? Or a name that is explicit? I really think it's misleading to use the word "validation" to mean "parse or validation". Words have meanings.